src/beliefpropagation.jl

```diff
@@ -7,35 +7,60 @@ function construct_initial_mts(
 end
 
 function construct_initial_mts(
-  tn::ITensorNetwork, subgraphs::DataGraph; init=(I...) -> @compat allequal(I) ? 1 : 0
+  tn::ITensorNetwork,
+  subgraphs::DataGraph;
```

Comment:
Do we need to pass both the tensor network and the subgraphs, or are the subgraphs sufficient?

Reply:
Also I would rename this function something like `message_tensors`. For two reasons:
1. Using "initial" implies we are going to be using the message tensors that we construct to initialize a calculation, but we could be using them for another purpose.
2. Spelling out the name fully instead of using the shorthand "mt" for "message tensor".

Reply:
Good idea on the name. I believe we need the tensor network for getting the correct indices on the message tensors.

Reply:
I understand that in practice it may be easier given the tensor network, but in principle isn't all that information in the subgraphs, i.e. you can just look at the common indices between subgraphs? I think in fact the `partition` function may even output that information as the edge metadata, if I remember correctly.

Reply:
Okay let me took a look into that and see if it does have that info and I can use it

Reply:
Yeah you are right it's all in there so I have removed `tn` as an argument.